### PR TITLE
[SPARK-33298][CORE][FOLLOWUP] Add Unstable annotation to `FileCommitProtocol`

### PR DESCRIPTION
**Pull request by c21**
### What changes were proposed in this pull request?

This is the followup from https://github.com/apache/spark/pull/33012#discussion_r659440833, where we want to add `@Unstable` to `FileCommitProtocol`, to give people a better idea of API.

### Why are the changes needed?

Make it easier for people to follow and understand code. Clean up code.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing unit tests, as no real logic change.

### CODE REVIEW

**Comment by c21** [comment]
cc @HyukjinKwon and @cloud-fan please take a look when you have time, thanks.

File: core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala [hunk 1]
```diff
@@ -107,10 +111,9 @@ abstract class FileCommitProtocol extends Logging {
    * if a task is going to write out multiple files to the same dir. The file commit protocol only
    * guarantees that files written by different tasks will not conflict.
    *
-   * This API should be implemented and called, instead of
-   * [[newTaskTempFile(taskContest, dir, ext)]]. Provide a default implementation here to be
-   * backward compatible with custom [[FileCommitProtocol]] implementations before Spark 3.2.0.
+   * @since 3.2.0
    */
+  @Unstable
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala:116]: @c21 can we mark this whole class as `@Unstable`?

Comment (c21) [reply core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala]: @HyukjinKwon - ah I see, updated.

**Review by HyukjinKwon — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala [hunk 2]
```diff
@@ -41,7 +42,11 @@ import org.apache.spark.util.Utils
  *    (or abortTask if task failed).
  * 3. When all necessary tasks completed successfully, the driver calls commitJob. If the job
  *    failed to execute (e.g. too many failed tasks), the job should call abortJob.
+ *
+ * NOTE: this class is exposed as an API considering the usage of many downstream custom
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala:46]: no big deal but I would use `@note` instead.

Comment (c21) [reply core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala]: sure, updated.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by c21** [comment]
Thank you @HyukjinKwon and @dongjoon-hyun for review!

**Pull request closed without merging**